File: client/ArkhamDB.Scraper/Deck.cs
namespace ArkhamDB.Scraper
{
    internal class Deck
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string InvestigatorId { get; set; }
        public DateTime Published { get; set; }
        public int Favorites { get; set; }
        public int Likes { get; set; }
        public int Comments { get; set; }
        public string Version { get; set; }

        public Deck(int id, string name, string investigatorId, DateTime published, int likes, int favorite, int comments, string version)
        {
            Id = id;
            Name = name;
            InvestigatorId = investigatorId;
            Published = published;
            Favorites = favorite;
            Likes = likes;
            Comments = comments;
            Version = version;
        }
    }
}

File: src/models/deck.ts
export interface Deck {
    id: number;
    name: string;
    likes: number;
    favorites: number;
    comments: number;
    version: string;
    published: Date;
}

export function loadDecks(investigatorId: string): Promise<Deck[]> {
    return fetch(`${window.location.href}/${investigatorId}.json`)
        .then(
            (response) => {
                return response.json();
            }
        )
        .then(
            (decks: Deck[]) => {
                return decks.map(d => ({
                    ...d,
                    name: d.name.replaceAll('&quote;', '"'),
                    published: new Date(d.published)
                }));
            }
        )
        .catch(
            (error) => {
                console.warn(`No decks found for investigator: ${investigatorId}.`);
                return [];
            }
        );
}